src/components/request-form/request-form.js

```diff
@@ -0,0 +1,155 @@
+import image from '../../assets/settings.png';
+import 'brace/mode/javascript';
+import 'brace/mode/json';
+import '../theme';
+
+const Container = styled.div`
+  font-family: ${defaultFontFamily};
+  font-style: ${defaultFontWeight};
+  font-size: ${defaultFomtSize};
+  width: ${(props) => props.width}px;
+  height: ${(props) => props.height}px;
+  background: #3f3f3f;
+  color: #767676;
+  min-width: 150px;
+`;
+const Column = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 5px;
+`;
+const Header = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+`;
+const Footer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: flex-end;
+`;
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+`;
+const Line = styled.div`
+  height: 1px;
+  border-bottom: 1px dashed #767676;
+  width: 100%;
+`;
+const Image = styled.img`
+  padding-right: 5px;
+  width: 16px;
+  height: 16px;
+`;
+const Title = styled.div`
+  text-transform: uppercase;
+  color: ${(props) => props.color};
+`;
+
+const argumentsCount = [{ label: 'One' }, { label: 'Two' }, { label: 'Three' }];
+const languages = [{ label: 'javascript' }, { label: 'json' }];
+
+export default class RequestForm extends React.Component {
+  static propTypes = {
+    width: PropTypes.number.isRequired,
+    height: PropTypes.number.isRequired,
+    path: PropTypes.string.isRequired,
+    input: PropTypes.string,
+    selectLanguage: PropTypes.func.isRequired,
+    setArgument: PropTypes.func.isRequired,
+    submit: PropTypes.func.isRequired,
+  };
+
+  state = {
+    language: 'javascript',
+    arguments: [''],
+  };
+
+  onLoad = (editor) => (this.editor = editor);
+  onChangeLanguage = ({ label }) => {
+    this.setState({ language: label });
+    this.editor.getSession().setMode(`ace/mode/${label}`);
+    this.props.selectLanguage(label);
+  };
+  onChangeArgumentsCount = ({ label }) => {
+    const args = this.state.arguments;
+    const reset = (n, array = []) => [...array, ...new Array(n).fill('')].slice(0, n);
+
+    switch (label) {
+      case 'One':
+        return this.setState({ arguments: reset(1, args) });
+      case 'Two':
+        return this.setState({ arguments: reset(2, args) });
+      case 'Three':
+        return this.setState({ arguments: reset(3, args) });
+      default:
+        return this.setState({ arguments: reset(1, args) });
+    }
+  };
+  onChangeInput = (index) => (newArgument) => {
+    const args = [...this.state.arguments];
+    args[index] = newArgument;
+    this.setState({ arguments: args });
+    this.props.setArgument(index, this.state.arguments);
+  };
+  onSubmit = () => {
+    const { language } = this.state;
+    const args =
+      language === 'javascript' ? this.state.arguments.map(JSONL.parse).map(JSON.stringify) : this.state.arguments;
```

Comment:
you can also destruct **arguments** from state

Reply:
suggest not to use magic string 'javascript' and put it in const

Reply:
Why do you need to firstly parse and then stringify the same values?

Reply:
first: arguments - required by js word.
second: because I want to convert js literal object to JSON. `{a: 1} -> {"a": 1}`

Reply:
You can see that in first mapper I use `JSONL.parse` not `JSON.parse`

Reply:
`L` - literal

Reply:
You're right